script_library/deployment-actions-openstack.sh

```diff
@@ -63,7 +63,6 @@ function clean_openstack(){
     run_ansible ${socok8s_absolute_dir}/playbooks/openstack-osh_instance.yml -e osh_node_delete=True
     echo "Delete CaaSP 3 nodes"
     run_ansible ${socok8s_absolute_dir}/playbooks/openstack-delete_caasp.yml
-    clean_caasp4
```

Comment:
Not sure how you did run into the problem you describe in the commit message. There is already a check that the terraform binary is available (see https://github.com/SUSE-Cloud/socok8s/blob/master/script_library/deployment-actions-openstack.sh#L55 ). Also I would really like to keep the caasp4 cleanup for people who deploy caasp4. Otherwise we waste resources in case the deployment is on the engcloud

Reply:
The check for the terraform binary only occurs during a caasp4 deployment. If we have deployed a caasp3 stack, we may not have terraform installed.

Reply:
I still don't get it - the `clean_caasp4` function does the check if the terraform binary is available. So we have 2 cases:

- If the binary is not available, the function `clean_caasp4` does noting. 
- I the binary is available, `playbooks/openstack-delete_caasp4.yml` is called which triggers `playbooks/roles/caasp4/tasks/cleanup_nodes.yml` which should do nothing if no terraform workspace is found.

Am I missing something?

Reply:
If the binary is not available, then the failed check will exit the script before we delete the SES node and the network stack.